clients/drcachesim/tests/raw2trace_unit_tests.cpp

```diff
@@ -1984,6 +2060,105 @@ test_rseq_side_exit_inverted_with_timestamp(void *drcontext)
         check_entry(entries, idx, TRACE_TYPE_FOOTER, -1));
 }
 
+/* Tests rseq side exit with timestamps. */
+bool
+test_rseq_side_exit_with_timestamps(void *drcontext)
+{
+    std::cerr << "\n===============\nTesting rseq side exit with timestamps\n";
+    instrlist_t *ilist = instrlist_create(drcontext);
+    // raw2trace doesn't like offsets of 0 so we shift with a nop.
+    instr_t *nop = XINST_CREATE_nop(drcontext);
+    instr_t *move1 =
+        XINST_CREATE_move(drcontext, opnd_create_reg(REG1), opnd_create_reg(REG2));
+    instr_t *move2 =
+        XINST_CREATE_move(drcontext, opnd_create_reg(REG2), opnd_create_reg(REG1));
+    instr_t *move3 =
+        XINST_CREATE_move(drcontext, opnd_create_reg(REG1), opnd_create_reg(REG2));
+    instr_t *move4 =
+        XINST_CREATE_move(drcontext, opnd_create_reg(REG1), opnd_create_reg(REG2));
+    instr_t *jmp1 = XINST_CREATE_jump(drcontext, opnd_create_instr(move4));
+    instr_t *jcc = XINST_CREATE_jump_cond(drcontext, DR_PRED_EQ, opnd_create_instr(jmp1));
+    instr_t *store =
+        XINST_CREATE_store(drcontext, OPND_CREATE_MEMPTR(REG2, 0), opnd_create_reg(REG1));
+    instrlist_append(ilist, nop);
+    instrlist_append(ilist, jcc);
+    instrlist_append(ilist, move1);
+    instrlist_append(ilist, store);
+    instrlist_append(ilist, move2);
+    instrlist_append(ilist, jmp1);
+    instrlist_append(ilist, move3);
+    instrlist_append(ilist, move4);
+    size_t offs_nop = 0;
+    size_t offs_jcc = offs_nop + instr_length(drcontext, nop);
+    size_t offs_move1 = offs_jcc + instr_length(drcontext, jcc);
+    size_t offs_store = offs_move1 + instr_length(drcontext, move1);
+    size_t offs_move2 = offs_store + instr_length(drcontext, store);
+    size_t offs_jmp1 = offs_move2 + instr_length(drcontext, move2);
+    size_t offs_move3 = offs_jmp1 + instr_length(drcontext, jmp1);
+    size_t offs_move4 = offs_move3 + instr_length(drcontext, move3);
+
+    std::vector<offline_entry_t> raw;
+    raw.push_back(make_header());
+    raw.push_back(make_tid());
+    raw.push_back(make_pid());
+    raw.push_back(make_line_size());
+    raw.push_back(make_timestamp(/*value=*/1));
+    raw.push_back(make_core());
+    raw.push_back(make_marker(TRACE_MARKER_TYPE_RSEQ_ENTRY, offs_move2));
+    raw.push_back(make_block(offs_jcc, /*instr_count=*/1));
+    raw.push_back(make_block(offs_move1, /*instr_count=*/2));
+    raw.push_back(make_memref(/*addr=*/42));
+    raw.push_back(make_block(offs_jmp1, /*instr_count=*/1));
```

Comment:
Add some comments explaining what happened: the jcc actually skipped move1;store;move2?

Reply:
done.